query/logicalplan/builder.go

```diff
@@ -135,6 +135,9 @@ func (b Builder) Aggregate(
 	}
 }
 
-func (b Builder) Build() *LogicalPlan {
-	return b.plan
+func (b Builder) Build() (*LogicalPlan, error) {
+	if err := NewValidator(b.plan).Validate(); err != nil {
```

Comment:
Agreed. The constructor doesn't do more that initialize the struct.

Reply:
Good call @thorfour & @metalmatze that's fixed now 😃 

At one point I was thinking we might need to keep some state during validation and it would be easier if could keep that state on the validator struct. That turned out not to be the case, so having this struct (and its constructor) is not really necessary.